jdaviz/tests/test_subsets.py

```diff
@@ -70,17 +59,12 @@ def test_region_from_subset_profile(jdaviz_app, spectral_cube_wcs):
     data = Data(flux=np.ones((256, 128, 128)), label='Test 1D Flux', coords=spectral_cube_wcs)
     jdaviz_app.data_collection.append(data)
 
-    subset_state = RoiSubsetState(data.pixel_component_ids[1],
-                                  data.pixel_component_ids[0],
-                                  XRangeROI(1, 3.5))
-
     jdaviz_app.add_data_to_viewer('spectrum-viewer', 'Test 1D Flux')
 
-    jdaviz_app.data_collection.new_subset_group(
-        subset_state=subset_state, label='rectangular')
+    jdaviz_app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(1, 3.5))
```

Comment:
I do wonder about this approach of creating an roi through a viewer. It matches our UI paradigm (as you can't define an roi without a viewer), but that's more of a context thing. Ideally, through code, you wouldn't need a viewer in order to define a ROI that is supposed to be viewer-independent, right? I'm surprised you can't go even deeper into Glue and define the ROI even lower

Reply:
The point was specifically to go less deep into the stack, I wanted to do it in a way that replicates what happens when you use a region selection tool in a viewer.

Reply:
For example, this way it matters what the Subset selection state (e.g. "Create new" vs "Subset 1" selected) is rather than injecting the subset deeper with a label. Later in the test I trigger the switch back to "Create new" that Pey Lian added by selecting another tool, so we get to make sure that's working as a bonus.

Reply:
> Pey Lian added by selecting another tool

To be fair... I _tried_ to implement it, but @astrofrog ended up rewriting everything anyway in my PR. So the credit really goes to him... 😅

Reply:
Ah, well, nevertheless!